Add "manual backup" instruction

y-popov opened the pull request:
https://jira.sanger.ac.uk/browse/HSW-170

y-popov committed:
Add "manual backup" instruction

y-popov committed:
Improve SQL error handling

y-popov committed:
Move validation into entry parsing

sb10 reviewed (changes requested):
I haven't looked through this codebase or seen what kind of tests there are, but there are at least some test files, and ideally this should have been developed with TDD and thus new tests added or existing tests for the instructions updated.

One of the benefits of HTMX is that it becomes possible to test the backend and frontend logic properly and in one place.

If there are no tests on the produced HTML and no tests of this kind, and that would be a big task, create a JIRA card about adding full testing for everything and I can approve this PR as-is. But if adding the tests for this is quick and simple, please add them.

y-popov committed:
Add tests for ManualBackup

y-popov committed:
Fix test

y-popov commented:
@sb10 that's a good point. 

There is now one universal function `ParseInstruction` responsible for validation input instructions. I added test to check that it can now accept a new ManualBackup instruction. 

I also changed the function `createTestEntries` to generate entries with all possible instruction values. Now all tests that work with functions requiring entries as input, now ensure that those function work with newly added ManualBackup. Thanks to that I found a bug in `ConvertCsvToMySQL`.

There is already a number of tests that test server functions that produce html. They are in `server/server_test.go`
